// instagram-filters/lib/presets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _a;
var filters_1 = require("./filters");
var pipe_1 = require("./utils/pipe");
/** Clarendon: adds light to lighter areas and dark to darker areas */
function clarendon() {
    return pipe_1.pipe(filters_1.brightness(0.1), filters_1.contrast(0.1), filters_1.saturation(0.15));
}
exports.clarendon = clarendon;
/** Gingham: Vintage-inspired, taking some color out */
function gingham() {
    return pipe_1.pipe(filters_1.sepia(0.04), filters_1.contrast(-0.15));
}
exports.gingham = gingham;
/** Moon: B/W, increase brightness and decrease contrast */
function moon() {
    return pipe_1.pipe(filters_1.grayscale(), filters_1.contrast(-0.04));
}
exports.moon = moon;
/** Lark: Brightens and intensifies colours but not red hues */
function lark() {
    return pipe_1.pipe(filters_1.brightness(0.08), filters_1.rgbAdjust([1, 1.03, 1.05]));
}
exports.lark = lark;
/** Reyes: a new vintage filter, gives your photos a “dusty” look */
function reyes() {
    return pipe_1.pipe(filters_1.sepia(0.4), filters_1.brightness(0.13), filters_1.contrast(-0.05));
}
exports.reyes = reyes;
/** Juno: Brightens colors, and intensifies red and yellow hues */
function juno() {
    return pipe_1.pipe(filters_1.rgbAdjust([1.01, 1.04, 1]), filters_1.saturation(0.3));
}
exports.juno = juno;
/** Slumber: Desaturates the image as well as adds haze for a retro, dreamy look – with an emphasis on blacks and blues */
function slumber() {
    return pipe_1.pipe(filters_1.brightness(0.1), filters_1.saturation(-0.5));
}
exports.slumber = slumber;
/** Crema: Adds a creamy look that both warms and cools the image */
function crema() {
    return pipe_1.pipe(filters_1.rgbAdjust([1.04, 1, 1.02]));
}
exports.crema = crema;
/** Ludwig: A slight hint of desaturation that also enhances light */
function ludwig() {
    return pipe_1.pipe(filters_1.brightness(0.05), filters_1.saturation(-0.03));
}
exports.ludwig = ludwig;
/** Aden: This filter gives a blue/pink natural look */
function aden() {
    return pipe_1.pipe(filters_1.colorFilter([228, 130, 225, 0.13]), filters_1.saturation(-0.2));
}
exports.aden = aden;
/** Perpetua: Adding a pastel look, this filter is ideal for portraits */
function perpetua() {
    return pipe_1.pipe(filters_1.rgbAdjust([1.05, 1.1, 1]));
}
exports.perpetua = perpetua;
/** Amaro: Adds light to an image, with the focus on the centre */
function amaro() {
    return pipe_1.pipe(filters_1.saturation(0.3), filters_1.brightness(0.15));
}
exports.amaro = amaro;
/** Mayfair: Applies a warm pink tone, subtle vignetting to brighten the photograph center and a thin black border */
function mayfair() {
    return pipe_1.pipe(filters_1.colorFilter([230, 115, 105, 0.05]), filters_1.saturation(0.15));
}
exports.mayfair = mayfair;
/** Rise: Adds a "glow" to the image, with softer lighting of the subject */
function rise() {
    return pipe_1.pipe(filters_1.colorFilter([255, 170, 0, 0.1]), filters_1.brightness(0.09), filters_1.saturation(0.1));
}
exports.rise = rise;
/** Hudson: Creates an "icy" illusion with heightened shadows, cool tint and dodged center */
function hudson() {
    return pipe_1.pipe(filters_1.rgbAdjust([1, 1, 1.25]), filters_1.contrast(0.1), filters_1.brightness(0.15));
}
exports.hudson = hudson;
/** Valencia: Fades the image by increasing exposure and warming the colors, to give it an antique feel */
function valencia() {
    return pipe_1.pipe(filters_1.colorFilter([255, 225, 80, 0.08]), filters_1.saturation(0.1), filters_1.contrast(0.05));
}
exports.valencia = valencia;
/** X-Pro II: Increases color vibrance with a golden tint, high contrast and slight vignette added to the edges */
function xpro2() {
    return pipe_1.pipe(filters_1.colorFilter([255, 255, 0, 0.07]), filters_1.saturation(0.2), filters_1.contrast(0.15));
}
exports.xpro2 = xpro2;
/** Sierra: Gives a faded, softer look */
function sierra() {
    return pipe_1.pipe(filters_1.contrast(-0.15), filters_1.saturation(0.1));
}
exports.sierra = sierra;
/** Willow: A monochromatic filter with subtle purple tones and a translucent white border */
function willow() {
    return pipe_1.pipe(filters_1.grayscale(), filters_1.colorFilter([100, 28, 210, 0.03]), filters_1.brightness(0.1));
}
exports.willow = willow;
/** Lo-Fi: Enriches color and adds strong shadows through the use of saturation and "warming" the temperature */
function lofi() {
    return pipe_1.pipe(filters_1.contrast(0.15), filters_1.saturation(0.2));
}
exports.lofi = lofi;
/** Inkwell: Direct shift to black and white */
function inkwell() {
    return pipe_1.pipe(filters_1.grayscale());
}
exports.inkwell = inkwell;
/** Hefe: Hight contrast and saturation, with a similar effect to Lo-Fi but not quite as dramatic */
function hefe() {
    return pipe_1.pipe(filters_1.contrast(0.1), filters_1.saturation(0.15));
}
exports.hefe = hefe;
/** Nashville: Warms the temperature, lowers contrast and increases exposure to give a light "pink" tint – making it feel "nostalgic" */
function nashville() {
    return pipe_1.pipe(filters_1.colorFilter([220, 115, 188, 0.12]), filters_1.contrast(-0.05));
}
exports.nashville = nashville;
/** Stinson: washing out the colors ever so slightly */
function stinson() {
    return pipe_1.pipe(filters_1.brightness(0.1), filters_1.sepia(0.3));
}
exports.stinson = stinson;
/** Vesper: adds a yellow tint that */
function vesper() {
    return pipe_1.pipe(filters_1.colorFilter([255, 225, 0, 0.05]), filters_1.brightness(0.06), filters_1.contrast(0.06));
}
exports.vesper = vesper;
/** Earlybird: Gives an older look with a sepia tint and warm temperature */
function earlybird() {
    return pipe_1.pipe(filters_1.colorFilter([255, 165, 40, 0.2]));
}
exports.earlybird = earlybird;
/** Brannan: Increases contrast and exposure and adds a metallic tint */
function brannan() {
    return pipe_1.pipe(filters_1.contrast(0.2), filters_1.colorFilter([140, 10, 185, 0.1]));
}
exports.brannan = brannan;
/** Sutro: Burns photo edges, increases highlights and shadows dramatically with a focus on purple and brown colors */
function sutro() {
    return pipe_1.pipe(filters_1.brightness(-0.1), filters_1.saturation(-0.1));
}
exports.sutro = sutro;
/** Toaster: Ages the image by "burning" the centre and adds a dramatic vignette */
function toaster() {
    return pipe_1.pipe(filters_1.sepia(0.1), filters_1.colorFilter([255, 145, 0, 0.2]));
}
exports.toaster = toaster;
/** Walden: Increases exposure and adds a yellow tint */
function walden() {
    return pipe_1.pipe(filters_1.brightness(0.1), filters_1.colorFilter([255, 255, 0, 0.2]));
}
exports.walden = walden;
/** 1977: The increased exposure with a red tint gives the photograph a rosy, brighter, faded look. */
function year1977() {
    return pipe_1.pipe(filters_1.colorFilter([255, 25, 0, 0.15]), filters_1.brightness(0.1));
}
exports.year1977 = year1977;
/** Kelvin: Increases saturation and temperature to give it a radiant "glow" */
function kelvin() {
    return pipe_1.pipe(filters_1.colorFilter([255, 140, 0, 0.1]), filters_1.rgbAdjust([1.15, 1.05, 1]));
}
exports.kelvin = kelvin;
/** Maven: darkens images, increases shadows, and adds a slightly yellow tint overal */
function maven() {
    return pipe_1.pipe(filters_1.colorFilter([225, 240, 0, 0.1]), filters_1.saturation(0.25), filters_1.contrast(0.05));
}
exports.maven = maven;
/** Ginza: brightens and adds a warm glow */
function ginza() {
    return pipe_1.pipe(filters_1.sepia(0.06), filters_1.brightness(0.1));
}
exports.ginza = ginza;
/** Skyline: brightens to the image pop */
function skyline() {
    return pipe_1.pipe(filters_1.saturation(0.35), filters_1.brightness(0.1));
}
exports.skyline = skyline;
/** Dogpatch: increases the contrast, while washing out the lighter colors */
function dogpatch() {
    return pipe_1.pipe(filters_1.contrast(0.15), filters_1.brightness(0.1));
}
exports.dogpatch = dogpatch;
/** Brooklyn */
function brooklyn() {
    return pipe_1.pipe(filters_1.colorFilter([25, 240, 252, 0.05]), filters_1.sepia(0.3));
}
exports.brooklyn = brooklyn;
/** Helena: adds an orange and teal vibe */
function helena() {
    return pipe_1.pipe(filters_1.colorFilter([208, 208, 86, 0.2]), filters_1.contrast(0.15));
}
exports.helena = helena;
/** Ashby: gives images a great golden glow and a subtle vintage feel */
function ashby() {
    return pipe_1.pipe(filters_1.colorFilter([255, 160, 25, 0.1]), filters_1.brightness(0.1));
}
exports.ashby = ashby;
/** Charmes: a high contrast filter, warming up colors in your image with a red tint */
function charmes() {
    return pipe_1.pipe(filters_1.colorFilter([255, 50, 80, 0.12]), filters_1.contrast(0.05));
}
exports.charmes = charmes;
exports.presetsMapping = (_a = {
        Clarendon: clarendon,
        Gingham: gingham,
        Moon: moon,
        Lark: lark,
        Reyes: reyes,
        Juno: juno,
        Slumber: slumber,
        Crema: crema,
        Ludwig: ludwig,
        Aden: aden,
        Perpetua: perpetua,
        Amaro: amaro,
        Mayfair: mayfair,
        Rise: rise,
        Hudson: hudson,
        Valencia: valencia
    },
    _a['X-Pro II'] = xpro2,
    _a.Sierra = sierra,
    _a.Willow = willow,
    _a['Lo-Fi'] = lofi,
    _a.Inkwell = inkwell,
    _a.Hefe = hefe,
    _a.Nashville = nashville,
    _a.Stinson = stinson,
    _a.Vesper = vesper,
    _a.Earlybird = earlybird,
    _a.Brannan = brannan,
    _a.Sutro = sutro,
    _a.Toaster = toaster,
    _a.Walden = walden,
    _a['1977'] = year1977,
    _a.Kelvin = kelvin,
    _a.Maven = maven,
    _a.Ginza = ginza,
    _a.Skyline = skyline,
    _a.Dogpatch = dogpatch,
    _a.Brooklyn = brooklyn,
    _a.Helena = helena,
    _a.Ashby = ashby,
    _a.Charmes = charmes,
    _a);
